### ISSUE TITLE
Define shell scripting style guide

### ISSUE THREAD

**Issue by tsenart** [mesosphere/kubernetes-mesos#429]
In order to have consistent, predictable and correct shell scripts, we should all follow the same guidelines. At the moment we have none, so let's take inspiration in the following guidelines and agree on something that makes sense to us.

https://google-styleguide.googlecode.com/svn/trunk/shell.xml
https://github.com/bahamas10/bash-style-guide

/cc @jdef @karlkfi @sttts

**Comment by sttts** [comment]
What are the ones that k8s upstream uses? it seems they have a number of rules, compare the review comments of the docker-compose branch.

**Comment by tsenart** [comment]
We could look at what they're doing, but I'd like us to apply this style guide to every shell script we write, not just the ones related to Kubernetes. I'm not sure where such meta-issues would fit.

**Comment by sttts** [comment]
Yes, but in the k8s repo we are bound to their style. Which other shell scripts do we have (other than some small Makefile snippets)?

**Comment by tsenart** [comment]
We got the DCOS image scripts, or did we completely get rid of those?

**Comment by sttts** [comment]
No, not completely, but reduced a lot. Maybe we should spend more time in porting those to Go than to make them nice and tidy.

**Comment by tsenart** [comment]
+1 for Go.

**Comment by karlkfi** [comment]
bahamas10's style guide conflicts not only with the k8s conventions but with google's guide and my own sense of style.

Personally, I think if it's not automated then it's a waste of time enforcing it strictly. That said, it's worth being consistent. It should be required to be consistent within a file and encouraged to be consistent with existing files. Style can always be improved later. So it shouldn't hold up a review too long, unless it's unreadable or inconsistent.

**Comment by tsenart** [comment]
@karlkfi: I suggested the style guides for inspiration only. We should pick the things that make sense to us and write up our own style guide. Enforcement can be done at review time and will be less time consuming once everyone is used to it. In another issue, I also suggested we use http://shellcheck.net/about.html
